src/hotspot/share/gc/g1/g1CollectedHeap.hpp

```diff
@@ -1121,14 +1121,12 @@ class G1CollectedHeap : public CollectedHeap {
                                  size_t length,
                                  uint worker_id) const;
 
-  // Returns the HeapRegion that contains addr. addr must not be NULL.
-  template <class T>
-  inline HeapRegion* heap_region_containing(const T addr) const;
+  // Returns the HeapRegion that contains addr. addr must not be nullptr.
+  inline HeapRegion* heap_region_containing(const void* addr) const;
 
-  // Returns the HeapRegion that contains addr, or NULL if that is an uncommitted
-  // region. addr must not be NULL.
-  template <class T>
-  inline HeapRegion* heap_region_containing_or_null(const T addr) const;
+  // Returns the HeapRegion that contains addr, or nullptr if that is an uncommitted
+  // region. addr must not be nullptr.
+  inline HeapRegion* heap_region_containing_or_null(HeapWord* addr) const;
```

Comment:
Why does `heap_region_containing` take `void*` but `heap_region_containing_or_null` takes `HeapWord*`.  Seems like they should be the same (probably `void*` for both?).  And why did the `const` qualifier get dropped for the or_null variant?

Reply:
`heap_region_containing_or_null` only needs the `HeapWord*` variant - I thought to specialize them as much as possible; dropping of the const has been unintenional. I will fix and regularize these two methods. Thanks.